src/components/Card.js

```diff
@@ -26,13 +39,42 @@ export default class Card {
   }
 
   _handleLikeIcon() {
-    this._cardElement
-      .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
+    const cardId = this._cardData._id;
+    const likeButton = this._cardElement.querySelector(".card__like-button");
+    const isLiked = likeButton.classList.contains("card__like-button_active");
+
+    if (cardId.isLiked) {
+      api
+        .dislikeCard(cardId)
+        .then(() => {
+          likeButton.classList.remove("card__like-button_active");
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    } else {
+      api
+        .likeCard(cardId)
+        .then(() => {
+          likeButton.classList.add("card__like-button_active");
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
   }
 
   _handleDeleteCard() {
-    this._cardElement.remove();
+    const cardId = this._cardData._id;
+    api
```

Comment:
Needs correction: the card should not know about the api and the delete confirmation, all the logic should be in index.js

Also, you should create PopupWithConfirmation class as it described in the task and show loading state on the confirm button during the api request

Reply:
some guidance on how to go about this would be appreciated as well

Reply:
`PopupWithConfirmation` is a general class. It can be used for any confirmation, so it should not know anything about the cards or the api. This class should have a setter for `_handleFormSubmit`, not for card, so you can use a card in that handler in `index.js`.

```js
constructor(popupSelector) {
  super(popupSelector);
  this._form = this._modal.querySelector(".modal__form");
}

setAction(action) { 
  this._handleFormSubmit = action; 
} 

this._form.addEventListener("submit", (evt) => { 
  evt.preventDefault(); 
  this._handleFormSubmit(); 
})
```